lib/python/util.py

```diff
@@ -36,19 +36,35 @@ def list_files_recursively(dir, pattern):
     return matches
 
 def get_git_revision_script_url(file_path):
-    import os
-    import subprocess
     """
-    file_path is the local file path from a git repo
+    file_path is the local file path in a github repo
     returns the github url with the hash value of the current HEAD
+
+    Only handles `git config --get remote.origin.url` output in the form defined
+    in pattern variable
     """
-    curr_dir = os.getcwd()
-    os.chdir(os.path.dirname(file_path)) # need to chg dir to run git commands
+    import os
+    from git import Repo
+    import re
+
+    curr_dir = os.path.dirname(file_path)
+    while '.git' not in os.listdir(curr_dir):
+        curr_dir = os.path.abspath(os.path.join(curr_dir, os.pardir))
+        if curr_dir == os.path.sep:
+            raise Exception('Cannot find recursively a .git in parent directories')
+
+    repo_name      = os.path.basename(curr_dir)
+    repo           = Repo(curr_dir)
+    remote_org_url = repo.remotes.origin.url
+
+    pattern = 'git@github.com:(.*)/%s' % repo_name
+    regroup = re.search(pattern, remote_org_url)
```

Comment:
> for the output, and don't use it anywhere else.

I don't understand what you mean, this is the point, to only use it to put it in front of the repo name.